apps/emqx/src/emqx_trace/emqx_trace_formatter.erl

```diff
@@ -100,7 +109,11 @@ format_payload(Payload, text) when ?MAX_PAYLOAD_FORMAT_LIMIT(Payload) ->
     unicode:characters_to_list(Payload);
 format_payload(Payload, hex) when ?MAX_PAYLOAD_FORMAT_LIMIT(Payload) -> binary:encode_hex(Payload);
 format_payload(<<Part:?TRUNCATED_PAYLOAD_SIZE/binary, _/binary>> = Payload, Type) ->
-    emqx_packet:format_truncated_payload(Part, byte_size(Payload), Type).
+    emqx_packet:format_truncated_payload(Part, byte_size(Payload), Type);
+format_payload(Payload, _) ->
```

Comment:
if payload is a json-map, maybe it makes sense to format?

Reply:
Not 100% sure about this but I think it makes sense to keep it as a map in this case because I think this is what will be most useful for users.

Reply:
right, since it's returning Payload as is, it should be included in the output.